mixer/adapter/svcctrl/expiringLRUCache.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package svcctrl
+
+import (
+	"sync"
+	"time"
+
+	"github.com/hashicorp/golang-lru/simplelru"
+	"github.com/jonboulle/clockwork"
+)
+
+type (
+	cacheValue struct {
+		// insert timestamp
+		timestamp time.Time
```

Comment:
iirc, `time.Time` values -> allocations.  I wonder if we can use a millis_since_epoch or similar instead.

It *might* be more storage-efficient to store with the expiration time as now().UnixNano() + expiration, instead of keeping a ref to the addition time.Time.

It'd be worth some sort of benchmark.